### ISSUE TITLE
The assertion `.should('not.have.text', text)` still passes even when one of the elements found matches the given text

### ISSUE THREAD

**Issue by antonyfuentes** [cypress-io/cypress#18097]
### Current behavior

The assertion `should('not.have.text', text)` still passes even when one of the elements found matches the given text.

### Desired behavior

If one of the elements found matches the text from the assertion `should('not.have.text', text)`, then the assertion should fail.

### Test code to reproduce

The following test shows a green status even when one of the elements found matches the given text:

```js
it('reproduces the issue', () => {
  cy.visit('https://example.cypress.io/todo')
  cy.get('.todo-list li').should('not.have.text', 'Pay electric bill')
})
```

### Cypress Version

8.4.0

### Screenshot

![Screen Shot 2021-09-14 at 17 14 42](https://user-images.githubusercontent.com/16458076/133345997-e8f5fb64-a78c-4dcb-a86c-89a576b1f693.png)

**Comment by mrpicklez70** [comment]
This might be related to dependency versions. It seems to have happened before: https://github.com/cypress-io/cypress/issues/6384#issuecomment-584167241

**Comment by sainthkh** [comment]
After more research, I learned that this is the expected behavior because Cypress uses `chai-jquery` behind the scene.

When using with multiple elements, it concatenates the texts and asserts with it. In the case above, 'Pay electric bill' is compared with 'Pay electric billWalk the dog'. 

I tested it with the code below with `chai-jquery`. And it also failed. 

```html
<html>
  <head>
    <title>Mocha</title>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    <script src="https://unpkg.com/jquery@3.6.0/dist/jquery.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/chai/4.3.4/chai.min.js"></script>
    <script src="https://unpkg.com/chai-jquery@2.1.0/chai-jquery.js"></script>
  </head>
  <body>
    <div>foo</div>
    <div>bar</div>
    <script>
      chai.expect($('div')).to.have.text('foo');
    </script>
  </body>
</html>
```

You can solve the problem by writing code with `each` below:

```js
it('reproduces the issue', () => {
  cy.visit('https://example.cypress.io/todo')
  cy.get('.todo-list li').each(($el) => {
    cy.wrap($el).should('not.have.text', 'Pay electric bill')
  })
})
```

Finally, I'm currently wondering if we should change the behavior of `have.text` to [the documentation](https://github.com/chaijs/chai-jquery#texttext):

> Assert that the text of the first element of the selection is equal to the given text

Currently, `chai-jquery` uses the combined version of text, not the first element.

**Comment by sainthkh** [comment]
I tried to fix this in #18504, but decided not to fix this and respect the current `chai-jquery` behavior. 

For some projects, this can be a huge breaking change. There can be some valid cases with this behavior like when you're checking some elements with consecutive `<span>`s.  

Maybe I will come back to this issue when `chai-jquery` behavior is fixed.
